src/utils/bytesParsers/formatBytes.ts

```diff
@@ -0,0 +1,131 @@
+import {formatNumber} from '..';
+import {GIGABYTE, KILOBYTE, MEGABYTE, TERABYTE} from '../constants';
+import {isNumeric} from '../utils';
+
+import i18n from './i18n';
+
+const sizes = {
+    b: {
+        value: 1,
+        label: i18n('b'),
+    },
+    kb: {
+        value: KILOBYTE,
+        label: i18n('kb'),
+    },
+    mb: {
+        value: MEGABYTE,
+        label: i18n('mb'),
+    },
+    gb: {
+        value: GIGABYTE,
+        label: i18n('gb'),
+    },
+    tb: {
+        value: TERABYTE,
+        label: i18n('tb'),
+    },
+};
+
+export type BytesSizes = keyof typeof sizes;
+
+interface FormatToSizeArgs {
+    value: number;
+    size?: BytesSizes;
+    precision?: number;
+    withLabel?: boolean;
+    isSpeed?: boolean;
+}
+
+const formatToSize = ({
+    value,
+    size = 'mb',
+    precision = 0,
+    withLabel = true,
+    isSpeed = false,
+}: FormatToSizeArgs) => {
+    let result = (Number(value) / sizes[size].value).toFixed(precision);
+    result = formatNumber(result);
+
+    if (withLabel) {
```

Comment:
Lets move this case and case with `isSpeed` to another functions. Now it's pretty unclear what for this function is responsible.

Reply:
Do you suggest separating conditions? Or there shouldn't be logic with adding label in such function?

Reply:
I see it as two additional functions: one will give simple label, one will give `/s` label.

```
function withSpeed(val) {
return withLabel(val)
}
function withLabel(val) {
return formatBytes(val)
}
```